forte/data/multi_pack.py

```diff
@@ -176,6 +176,172 @@ def get_span_text(self, begin: int, end: int):
             "specific data pack to get text."
         )
 
+    def remove_pack(
+        self, index_of_pack: int, clean_invalid_entries: bool = False
+    ) -> bool:
+        """
+        Remove a data pack at index `index_of_pack` from this multi pack.
+
+        In a multi pack, the data pack to be removed may be associated with
+        some multi pack entries, such as `MultiPackLinks` that are connected
+        with other packs. These entries will become dangling and invalid,
+        thus need to be removed. One can consider removing these links before
+        calling this function, or set the `clean_invalid_entries` to `True` so
+        that they will be automatically pruned.
+
+        Args:
+            index_of_pack (int): The index of pack for removal from the
+              multi pack. If invalid, no pack will be deleted.
+            clean_invalid_entries (bool): Switch for automatically cleaning
+              the entries associated with the data pack being deleted which
+              will become invalid after the removal of the pack. Default is
+              False.
```

Comment:
Can we add an argument to purge the pack list in this function? With the newer pack id mechanism, it should be safer to do this now.